src/run/allocator.rs

```diff
@@ -0,0 +1,100 @@
+use super::*;
+
+/// The memory behind a two-word allocation.
+///
+/// This must be aligned to 16 bytes so that the left word's address always ends
+/// with `0b0000` and the right word's address always ends with `0b1000`.
+#[repr(C)]
+#[repr(align(16))]
+#[derive(Default)]
+pub(super) struct Node(pub AtomicU64, pub AtomicU64);
+
+#[repr(align(16))]
+pub struct Heap(pub(super) [Node]);
+
+impl Heap {
+  /// Allocate an area for the net's heap with a given size.
+  pub fn new(size: usize) -> Box<Heap> {
```

Comment:
Removing `Net::init_heap` is a backwards-incompatible change; we are going to have to coordinate with the lang team, because I'm pretty sure they use it somewhere.

Reply:
Yeah, there are a few breaking changes to the API here.

Now that the circular dependency is removed, I don't think it needs to be coördinated very much; they can simply update `hvm-core` once they're ready.